worker/buildbot_worker/commands/transfer.py

```diff
@@ -107,7 +107,8 @@ def start(self):
         self._reactor.callLater(0, self._loop, d)
 
         def _close_ok(res):
-            self.fp = None
+            if self.fp:
+                self.fp.close()
```

Comment:
Also on line 98, in the `except` block where self.fp is set to `None` do we need to close self.fp?
Maybe at the beginning of the function it would be good to initialize self.fp to None before
calling open(), just to make our intention clear.